### PR TITLE
Task 11 CORE-GET/api/articles-sorting-queries...

### PR DESCRIPTION
**Pull request by filosoho**
...FEATURE REQUEST the endpoint accepts queries sort_by and order

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -122,6 +122,73 @@ describe("/api/articles", () => {
...
+
+    test("GET 200: ?sort_by=created_at&order=asc returns articles sorted by created_at in ascending order", () => {
+      return request(app)
+        .get("/api/articles?sort_by=created_at&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBeGreaterThan(0);
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: false,
+          });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:134]: Interesting way to check that it is ascending! I quite like it :D I think I'd also check that ascending is true though :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -122,6 +122,73 @@ describe("/api/articles", () => {
...
+        .get("/api/articles?sort_by=comment_count&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBeGreaterThan(0);
+          expect(body.articles).toBeSortedBy("comment_count", {
+            descending: false,
+          });
+        });
+    });
+
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:159]: I'm not sure that this test needed to happen ^^ you've already proven that a valid column can be sorted by asc :)

File: models/articles.model.js [hunk 3]
```diff
@@ -1,27 +1,45 @@
...
+    "article_id",
+    "title",
+    "author",
+    "body",
+    "topic",
+    "created_at",
+    "votes",
+    "comment_count",
+  ];
+  const validOrder = ["asc", "desc", "ASC", "DESC"];
```
Comment (nc-niamh-smith) [inline models/articles.model.js:16]: Is there something we could do to the string beforehand to make sure it's uppercase?

File: models/articles.model.js [hunk 4]
```diff
@@ -1,27 +1,45 @@
...
-      return articles;
-    })
-    .catch((err) => {
-      return Promise.reject(err);
-    });
+
+  return db.query(queryStr).then(({ rows }) => {
+    return rows.map((row) => ({
+      ...row,
+      comment_count: Number(row.comment_count),
```
Comment (nc-niamh-smith) [inline models/articles.model.js:40]: There is a way to ensure that this is a number when it is returned from the query! :) Have a look into the CAST sql command 😄

**Review by nc-niamh-smith — COMMENTED** [review]
A couple of notes, but nothing too hectic! 😄

File: models/articles.model.js (hunk 4 above)
Comment (filosoho) [reply models/articles.model.js:40]: Nice one Niamh! Thank you for pointing this out. :)

File: models/articles.model.js (hunk 3 above)
Comment (filosoho) [reply models/articles.model.js:16]: Good point! I did it further away in the code, but it makes sense to do it before this line. :) Thank you! Will change this. :)

File: __tests__/app.test.js (hunk 1 above)
Comment (filosoho) [reply __tests__/app.test.js:134]: Will add it! Thank you Niamh! :)

File: __tests__/app.test.js (hunk 2 above)
Comment (filosoho) [reply __tests__/app.test.js:159]: Actually, this made me realise that I didn't make sure that the sort_by and order resolves to defaults, if nothing is passed. I made an error handler for that instead. :sweat_smile: Gonna fix this in my next commit. :raised_hands: Thanks Niamh!

**Pull request merged**